### PR TITLE
fix: selection strategy madness when its not reset

### LINKED ISSUE #1538: `selection_strategy = "row"` is broken for `builtin.buffers()`
### Description

`selection_strategy = "row"` broken for `builtin.buffers()`

https://user-images.githubusercontent.com/19517248/144515397-05a5d3f4-2d2a-44b4-99c1-bd7e27c067d8.mp4

This used to work fine as shown here: https://github.com/nvim-telescope/telescope.nvim/pull/828

### Neovim version

```markdown
NVIM v0.6.0-dev+558-g3ba800f15
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

### Operating system and version

Manjaro Linux x86_64 -  Kernel: 5.10.79-1-MANJARO

### checkhealth telescope

```markdown
telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - WARNING: nvim-treesitter not found. 

## Checking external dependencies
  - OK: rg: found ripgrep 13.0.0
  - OK: fd: found fd 8.2.1
```

### Steps to reproduce

See video above.

### Expected behavior

The selection should stay on the same row after pressing "m".

### Actual behavior

The selection resets and goes to the first row after pressing "m".

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          { 'nvim-telescope/telescope-fzf-native.nvim', run = 'make' },
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup({defaults = {selection_strategy = "row"}})
  require('telescope').load_extension('fzf')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

### PR DESCRIPTION
**Pull request by Conni2461**
fix #1538 #1539

CC @mroavi would be cool if you could test this. Also for other regressions

### CODE REVIEW

File: lua/telescope/pickers.lua [hunk 1]
```diff
@@ -1143,13 +1143,6 @@ function Picker:get_result_processor(find_id, prompt, status_updater)
     end
 
     self.sorter:score(prompt, entry, cb_add, cb_filter)
-
-    -- Only on the first addition do we want to set the selection.
-    -- This allows us to handle moving the cursor to the bottom or top of the window
-    -- depending on the strategy.
-    if count == 1 then
-      self:_do_selection(prompt)
-    end
```
Comment (Conni2461) [inline lua/telescope/pickers.lua:1152]: @tjdevries are you trolling me. Laughing. That just forces selection_strategy `reset`, breaking `row`, ...

If we have reset set we will move to the bottom | top regardless. So why is it here? Everything seems to work without it.

**Comment by mroavi** [comment]
Deleting the last entry in the buffer's picker, makes the selection reset to the first entry. Would it be possible to handle that case so that it stays on the last entry?

File: lua/telescope/pickers.lua (hunk 1 above)
Comment (tjdevries) [reply lua/telescope/pickers.lua:1152]: LOL, just saw this. Who knows. This is a confusing landscape of confusion.

Comment (Conni2461) [reply lua/telescope/pickers.lua:1286]: :rofl: guess it wasn't important

**Comment by Conni2461** [comment]
Its fixed now. I am very sorry for the long wait.

**Comment by Conni2461** [comment]
was hotfixed by https://github.com/nvim-telescope/telescope.nvim/commit/df303e12e09b4e6e1fd1e1e184d26827105b318d

**Pull request merged**